opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1625,53 +1625,100 @@ namespace Opm
     template <typename TypeTag>
     void
     WellInterface<TypeTag>::
-    updateWellStateRates(const Simulator& simulator,
-                         WellState<Scalar>& well_state,
-                         DeferredLogger& deferred_logger) const
+    initializeWellState(const Simulator& simulator,
+                        const GroupState<Scalar>& group_state,
+                        WellState<Scalar>& well_state,
+                        DeferredLogger& deferred_logger) const
     {
         OPM_TIMEFUNCTION();
         // Check if the rates of this well only are single-phase, do nothing
         // if more than one nonzero rate.
         auto& ws = well_state.well(this->index_of_well_);
-        int nonzero_rate_index = -1;
-        const Scalar floating_point_error_epsilon = 1e-14;
-        for (int p = 0; p < this->number_of_phases_; ++p) {
-            if (std::abs(ws.surface_rates[p]) > floating_point_error_epsilon) {
-                if (nonzero_rate_index == -1) {
-                    nonzero_rate_index = p;
-                } else {
-                    // More than one nonzero rate.
-                    return;
-                }
-            }
-        }
-
+        // int nonzero_rate_index = -1;
+        // const Scalar floating_point_error_epsilon = 1e-14;
+        // for (int p = 0; p < this->number_of_phases_; ++p) {
+        //     if (std::abs(ws.surface_rates[p]) > floating_point_error_epsilon) {
+        //         if (nonzero_rate_index == -1) {
+        //             nonzero_rate_index = p;
+        //         } else {
+        //             // More than one nonzero rate.
+        //             return;
+        //         }
+        //     }
+        // }
         // Calculate the rates that follow from the current primary variables.
-        std::vector<Scalar> well_q_s = computeCurrentWellRates(simulator, deferred_logger);
-
-        if (nonzero_rate_index == -1) {
+        std::vector<Scalar> well_q_s;
+        Scalar bhp_tmp = 0.0;
```

Comment:
With this design, the function `initializeWellState()` can only be used with producers. I think it is used for both types of wells.